src/classes/IQueryFilter.cls

```diff
@@ -14,7 +14,7 @@ public interface IQueryFilter {
 
     // Setter methods
     IQueryFilter filterByField(Schema.SObjectField fieldToFilter, QueryOperator operator, Object providedValue);
-    IQueryFilter filterByParentField(List<Schema.SObjectField> fieldToFilterChain, QueryOperator operator, Object providedValue);
+    IQueryFilter filterByParentField(SObjectFieldWrapper fieldWrapper, QueryOperator operator, Object providedValue);
```

Comment:
yeah the only thing that is awkward is the initialization of the wrapper.  In this case, it may be better to signal intent (maintaining the two separate methods) and initialize the wrapper from within the filterByParentField signature.  Gives the callers less monitor width to have to traverse!